src/runtime/components/elements/Accordion.vue

```diff
@@ -108,12 +108,25 @@ export default defineComponent({
       default: () => ({})
     }
   },
-  setup (props) {
+  emits: ['open'],
+  setup (props, { emit }) {
     const { ui, attrs } = useUI('accordion', toRef(props, 'ui'), config, toRef(props, 'class'))
 
     const uiButton = computed<typeof configButton>(() => configButton)
 
     const buttonRefs = ref<{ open: boolean, close: (e: EventTarget) => {} }[]>([])
+    
+    const openedStates = computed(() => buttonRefs.value.map(({ open }) => open))
```

Comment:
I couldn't read directly the button refs in the watcher, because old and new value were identical, I don't know the reason. I had to use this intermediary state to effectively detect the changes

Reply:
Did you use a function in the watcher? 🤔

Reply:
Just passed the refs like this and the values were always equal

```ts
watch(buttonRefs, ...)
```

Reply:
Just an idea but what if you use `watch(() => buttonRefs.value, ...)`?

Reply:
Just tried, the exact same thing is happening. I wonder if this is not because the refs are still references to the dom and updated before the watcher's callback is ran, whereas the computed is only updated on the next tick

Reply:
That might be the case yes. I don't mind having a computed anyway.